ui/webui/src/components/Error.jsx

```diff
@@ -16,43 +16,131 @@
+    const queryData = {
+        ...productQueryData,
+        component: "anaconda",
+    };
+
+    const reportURL = new URL(baseURL);
+    reportURL.pathname = "enter_bug.cgi";
+    Object.keys(queryData).map(query => reportURL.searchParams.append(query, queryData[query]));
+    return reportURL.href;
+};
+
+const addExceptionDataToReportURL = (url, exception) => {
+    const newUrl = new URL(url);
+
+    const context = exception.contextData?.context ? exception.contextData?.context + " " : "";
+
+    newUrl.searchParams.append(
+        "short_desc",
+        "WebUI: " + context + exception.name + ": " + exception.message
+    );
+    newUrl.searchParams.append(
+        "comment",
+        "Installer WebUI Critical Error:\n" + context + exception.name + ": " + exception.message + "\n\n" + _("Please attach the file /tmp/webui.log to the issue.")
+    );
+    return newUrl.href;
+};
+
+export const CriticalError = ({ exception, isBootIso, reportLinkURL }) => {
+    const reportURL = addExceptionDataToReportURL(reportLinkURL, exception);
+    const [logContent, setLogContent] = useState("");
+
+    useEffect(() => {
+        cockpit.spawn(["journalctl", "-a"])
+                .then(content => setLogContent(content));
+    }, []);
+
+    const openBZIssue = (reportURL) => {
+        cockpit.file("/tmp/webui.log").replace(logContent)
+                .then(window.open(reportURL, "_blank", "noopener,noreferer"));
+    };
+
+    const context = exception.contextData?.context;
+
     return (
         <Modal
-          description={_("The installer cannot continue due to a critical error.")}
+          description={context
+              ? cockpit.format(_("The installer cannot continue due to a critical error: $0"), _(context))
+              : _("The installer cannot continue due to a critical error.")}
           id="critical-error-modal"
           isOpen
           position="top"
           showClose={false}
           title={_("Critical error")}
           titleIconVariant="danger"
-          variant="small"
+          variant={ModalVariant.large}
           footer={
               <>
+                  {reportLinkURL &&
+                  <Button
+                    variant="primary"
+                    icon={<ExternalLinkAltIcon />}
+                    onClick={() => openBZIssue(reportURL)}
+                    component="a">
+                      {_("Report issue")}
+                  </Button>}
                   <Button variant="secondary" onClick={exitGui}>
                       {isBootIso ? _("Reboot") : _("Quit")}
                   </Button>
               </>
           }>
-            <TextContent>
-                <Text component={TextVariants.p}>
-                    {exception.message}
-                </Text>
-            </TextContent>
+            <Form>
+                <FormGroup fieldId="critical-error-review-attached-log" label={_("Error details")}>
+                    <TextContent>
```

Comment:
the fieldid from the formgroup should be used as id here.

Reply:
Fixed